### ISSUE TITLE
allow folks to switch a build between local/private images to public ones via a single simple property

### ISSUE THREAD

**Issue by jstrachan** [rhuss/docker-maven-plugin#45]
so imagine you're working in development on an image that you wanna push to a registry so you can use it on multiple boxes (e.g. with kubernetes); so you might wanna use something like this...
```
    <docker.image>${env.DOCKER_REGISTRY}/mystuff/${project.artifactId}:${project.version}</docker.image>
```
then wherever your private docker registry is at $DOCKER_REGISTRY you're good to go. 

However then at some point you might wanna release these images to the global registry. Now we could replace the $DOCKER_REGISTRY with registry.index.docker.io; however the common convention is to omit a registry prefix if its the global one in docker image names. 

So we'd wanna switch from  ${env.DOCKER_REGISTRY}/mystuff/${project.artifactId}:${project.version} to just mystuff/${project.artifactId}:${project.version} when doing a release.

So I'm wondering is it maybe cleaner to not specify the registry in the image name; but use docker.registry instead? Then if docker.registry is specified and its not the public registry, the mvn plugins would prefix the image name with it; then do things the usual docker way - otherwise it would be omitted?

To do this on the fabric8 quickstarts I had to hack this a bit with profiles ;)

e.g. I defined a **docker.registryPrefix** property:
https://github.com/fabric8io/quickstarts/blob/master/apps/pom.xml#L39

which defaults to "${env.DOCKER_REGISTRY}/" which I then add to the front of docker names.

Then I use a profile (docker-public) to override this in each build....
https://github.com/fabric8io/quickstarts/blob/master/apps/pom.xml#L117

it works but it'd be kinda nicer to just specify the docker.registry (maybe on the CLI) and for that to work and do the right thing in terms of image name (omitting it from the image name if its a public image, or appending it for non-public registries)

**Comment by jgangemi** [comment]
is this still an open issue given you can override the registry on the command line using `-Ddocker.registry=...`?

**Comment by rhuss** [comment]
This is fixed and should work with >0.11.5-M1 : You can use `docker.registry`, an environment variable `$DOCKER_REGISTRY` or even a configuration in the image section (possibly merged in by a profile).
